dbms/src/Storages/KVStore/MultiRaft/RegionSerde.cpp

```diff
@@ -0,0 +1,272 @@
+/// 2. Vx -> V2 where x >= 2, in 7.5.0
+/// 3. Vx -> V2 where x >= 2, in later 7.5
+RegionPtr Region::deserialize(ReadBuffer & buf, const TiFlashRaftProxyHelper * proxy_helper)
+{
+    const auto binary_version = readBinary2<UInt32>(buf);
+    auto current_version = Region::CURRENT_VERSION;
+    fiu_do_on(FailPoints::force_region_read_version, {
+        if (auto v = FailPointHelper::getFailPointVal(FailPoints::force_region_read_version); v)
+        {
+            current_version = std::any_cast<UInt64>(v.value());
+            LOG_WARNING(
+                Logger::get(),
+                "Failpoint force_region_read_version set region binary version, value={}",
+                current_version);
+        }
+    });
+
+    if (current_version <= 1 && binary_version > current_version)
+    {
+        // Conform to https://github.com/pingcap/tiflash/blob/43f809fffde22d0af4c519be4546a5bf4dde30a2/dbms/src/Storages/KVStore/Region.cpp#L197
+        // When downgrade from x(where x > 1) -> 1, the old version will throw with "unexpected version".
+        // So we will also throw here.
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "Don't support downgrading from {} to {}",
+            binary_version,
+            current_version);
+    }
+    const auto binary_version_decoded = magic_enum::enum_cast<RegionPersistVersion>(binary_version);
+    if (!binary_version_decoded.has_value())
+    {
+        LOG_DEBUG(DB::Logger::get(), "Maybe downgrade from {} to {}", binary_version, current_version);
+    }
+
+    // Deserialize meta
+    RegionPtr region = std::make_shared<Region>(RegionMeta::deserialize(buf), proxy_helper);
+
+    // Try deserialize flag
+    if (binary_version >= 2)
+    {
+        auto flags = readBinary2<UInt32>(buf);
+        if ((flags & RegionPersistFormat::HAS_EAGER_TRUNCATE_INDEX) != 0)
+        {
+            region->eager_truncated_index = readBinary2<UInt64>(buf);
+        }
+        UInt32 extension_cnt = flags >> 1;
+        for (UInt32 i = 0; i < extension_cnt; i++)
+        {
+            auto [extension_type, length] = getPersistExtensionTypeAndLength(buf);
+            bool debug_continue = false;
+            using bundle_type = std::pair<int, std::shared_ptr<int>>;
+            fiu_do_on(FailPoints::force_region_read_extension_field, {
+                if (auto v = FailPointHelper::getFailPointVal(FailPoints::force_region_read_extension_field); v)
+                {
+                    auto bundle = std::any_cast<bundle_type>(v.value());
+                    if (bundle.first & 1)
+                    {
+                        if (extension_type == magic_enum::enum_underlying(RegionPersistExtension::TEST))
+                        {
+                            RUNTIME_CHECK(length == 4);
+                            RUNTIME_CHECK(readStringWithLength(buf, 4) == "abcd");
+                            debug_continue = true;
+                            *(bundle.second) += 1;
```

Comment:
```suggestion
                            *(bundle.second) += 1;
                            continue;
```
I think you can directly write "continue;" here rather than defining a `debug_continue` and apply "continue" outside the `fiu_do_on`?

Reply:
It won't work, will go to `Check false failed: Unhandled extension 2 length=4`

Reply:
I got it. Because there is a `do ... while(false)` in the macro `fiu_do_on()` :(